Source/Core/DolphinQt/Debugger/CodeDiffDialog.cpp

```diff
@@ -0,0 +1,383 @@
+                     [&symbol](Diff& v) { return v.symbol == symbol; }))
+    {
+      tmp_diff.symbol = symbol;
+      tmp_diff.addr = iter.addr;
+      tmp_diff.hits = iter.run_count;
+      current.push_back(tmp_diff);
+    }
+  }
+
+  // Could add address based difference instead of symbols. Probably need second function.
+  // Sort for lower_bound.
+  sort(current.begin(), current.end(),
+       [](const Diff& v1, const Diff& v2) { return (v1.symbol < v2.symbol); });
+
+  // If both lists are empty, write and skip.
+  if (!isize && !xsize)
+  {
+    if (include)
+      m_include = current;
+    else
+      m_exclude = current;
+    return;
+  }
+
+  // Update exclude list. Compare to exclude list if it exists and make current_diff to check
+  // against include.
+  if (!xsize && !include)
+  {
+    m_exclude = current;
+  }
+  else if (xsize)
+  {
+    for (auto& iter : current)
+    {
+      auto pos = lower_bound(m_exclude.begin(), m_exclude.end(), iter.symbol, AddrOP);
+
+      if (pos->symbol != iter.symbol)
+      {
+        current_diff.push_back(iter);
+
+        if (!include)
+          m_exclude.insert(pos, iter);
+      }
+    }
+    // If there is no include list, we're done.
+    if (!include && !isize)
+      return;
+  }
+  else
+  {
+    current_diff = current;
+  }
+
+  // Update include list.
+  if (include && isize)
+  {
+    // Compare include with current and remove items that aren't in both.
+    std::vector<Diff> tmp_swap;
+    for (auto& iter : m_include)
+    {
+      if (std::any_of(current_diff.begin(), current_diff.end(),
+                      [&](Diff& v) { return v.symbol == iter.symbol; }))
+        tmp_swap.push_back(iter);
+    }
+    m_include.swap(tmp_swap);
+  }
+  else if ((!isize && include) || (!xsize && !include))
+  {
+    // If both lists are about to be filled for the first time, compare full lists and remove from
+    // include list.
+    if (include)
+      m_include = current;
+
+    for (auto& list : m_exclude)
+    {
+      m_include.erase(std::remove_if(m_include.begin(), m_include.end(),
+                                     [&](Diff const& v) { return v.symbol == list.symbol; }),
+                      m_include.end());
+    }
+  }
+  else
+  {
+    // If exclude pressed, remove new exclude items from include list that match.
+    for (auto& list : current_diff)
+    {
+      m_include.erase(std::remove_if(m_include.begin(), m_include.end(),
+                                     [&](Diff& v) { return v.symbol == list.symbol; }),
+                      m_include.end());
+    }
+  }
+}
+
+void CodeDiffDialog::Update(bool include)
+{
+  // Wrap everything in a pause
+  Core::State old_state = Core::GetState();
+  if (old_state == Core::State::Running)
+    Core::SetState(Core::State::Paused);
+
+  // Main process
+  OnIncludeExclude(include);
+
+  m_matching_results_list->clear();
+
+  new QListWidgetItem(tr("Address\tHits\tSymbol"), m_matching_results_list);

```

Comment:
This free standing `new` surprised me, is it being used?

Reply:
This seems to be a quirk of Qt.
It appends to the QWidget parent, so without it we lose the Address\Hits\Symbol at the top of the result list

![removed-standing-new](https://user-images.githubusercontent.com/14857235/103445579-49322c00-4c33-11eb-87ff-d92466875fb7.png)

Reply:
This might be a lot more work but would it make sense to change `m_matching_results_list` to a table?  Then you could set the header with https://doc.qt.io/qt-5/qtablewidget.html#setVerticalHeaderLabels .  That would potentially open up the ability to sort by column in the future (not sure if that's useful?)

Reply:
As is, Dolphin still lacks this feature. I'd rather get this finally in master and then work on a refactor closer to Ultimap.

Still working on your other suggestions at the moment.

Reply:
Gotcha, that makes sense to me.

Reply:
Experimented with converting to QTableWidget. While it brings some advantages, it takes a lot of tweaking as a custom QTableWidget type to get a column to act as a single row while maintaining the properties of a QTableWidget. I'm going to stick with List for now.